types/tx.go

```diff
@@ -41,3 +45,19 @@ type TxProof struct {
 	Data     Tx               `json:"data"`
 	Proof    merkle.Proof     `json:"proof"`
 }
+
+func (txs Txs) ToTxsWithISRs(intermediateStateRoots IntermediateStateRoots) ([]pb.TxWithISRs, error) {
+	expectedLength := len(txs) + 2
+	if len(intermediateStateRoots.RawRootsList) != len(txs)+2 {
+		return nil, fmt.Errorf("invalid length of ISR list: %d, expected length: %d", len(intermediateStateRoots.RawRootsList), expectedLength)
+	}
+	txsWithISRs := make([]pb.TxWithISRs, 0)
```

Comment:
is it more efficient to do as below when u know the size: 
```
size := len(txs)
txsWithISRs := make([]pb.TxWithISRs, size, size)
```
and use direct access to assign:
```
txsWithISRs[i] = pb.TxWithISRs{
			PreIsr:  intermediateStateRoots.RawRootsList[i],
			Tx:      tx,
			PostIsr: intermediateStateRoots.RawRootsList[i+1],
		}
```

Reply:
yes it is. added.